src/sentry/static/sentry/app/views/settings/incidentRules/triggers/actionsPanel/index.tsx

```diff
@@ -306,45 +285,19 @@ class ActionsPanel extends React.PureComponent<Props> {
                     ) : (
                       <span />
                     )}
-
-                    {isUser || isTeam ? (
-                      <SelectMembers
-                        disabled={disabled}
-                        key={isTeam ? 'team' : 'member'}
-                        showTeam={isTeam}
-                        project={projects.find(({slug}) => slug === currentProject)}
-                        organization={organization}
-                        value={action.targetIdentifier}
-                        onChange={this.handleChangeTargetIdentifier.bind(
-                          this,
-                          triggerIndex,
-                          i
-                        )}
-                      />
-                    ) : availableAction?.inputType === 'select' ? (
-                      <SelectControl
-                        isDisabled={disabled || loading}
-                        value={action.targetIdentifier}
-                        options={availableAction?.options || []}
-                        onChange={this.handleChangeTargetIdentifier.bind(
-                          this,
-                          triggerIndex,
-                          i
-                        )}
-                      />
-                    ) : (
-                      <Input
-                        disabled={disabled}
-                        key={action.type}
-                        value={action.targetIdentifier}
-                        onChange={this.handleChangeSpecificTargetIdentifier.bind(
-                          this,
-                          triggerIndex,
-                          i
-                        )}
-                        placeholder={getPlaceholderForType(action.type)}
-                      />
-                    )}
+                    <ActionTargetSelector
```

Comment:
@mgaeta FYI, it would have been fine to just make a class method here to handle this logic so you don't have to bind and pass props down. But a separate component also works, thanks!